bootloader/grub.go

```diff
@@ -439,6 +443,24 @@ func (g *grub) CandidateCommandLine(pieces CommandLineComponents) (string, error
 	return g.commandLineForEdition(edition, pieces)
 }
 
+func (g *grub) DefaultCommandLine() (string, error) {
+	currentBootConfig := filepath.Join(g.dir(), "grub.cfg")
+
+	edition, err := editionFromDiskConfigAsset(currentBootConfig)
+	if err != nil {
+		if err != errNoEdition {
+			return "", fmt.Errorf("cannot obtain edition number of current boot config: %v", err)
+		}
+		// we were called using the TrustedAssetsBootloader interface
+		// meaning the caller expects to us to use the managed assets,
+		// since one on disk is not managed, use the initial edition of
+		// the internal boot asset which is compatible with grub.cfg
+		// used before we started writing out the files ourselves
+		edition = 1
```

Comment:
It was copied from `CommandLine`. To make sure they behave in a compatible way. I have no idea why `CommandLine` does this.

I think I should maybe refactor that code to avoid duplication of this. There is no point having the same "work-around" (or whatever this is) in 2 places.

Reply:
I have refactored the code here